### PR TITLE
Rename temp model to thermal model

### PR DESCRIPTION
**Pull request by 3d-gussner**
- [ ] Update https://reprap.org/wiki/G-code#M310:_Temperature_model_settings after PR has been merged

### CODE REVIEW

**Comment by gudnimg** [comment]
FYI @3d-gussner There are some merge conflicts now

**Comment by 3d-gussner** [comment]
@gudnimg  Yes there are always merge conflicts after the pre-release PRs have been merged.

File: Firmware/Marlin_main.cpp [hunk 1]
```diff
@@ -7311,9 +7311,9 @@ SERIAL_PROTOCOLPGM("\n\n");
     }
     break;
 
-#ifdef TEMP_MODEL
+#ifdef THERMAL_MODEL
     /*!
-    ### M310 - Temperature model settings <a href="https://reprap.org/wiki/G-code#M310:_Temperature_model_settings">M310: Temperature model settings</a>
+    ### M310 - Thermal model settings <a href="https://reprap.org/wiki/G-code#M310:_Themal_model_settings">M310: Thermal model settings</a>
```
Comment (wavexx) [inline Firmware/Marlin_main.cpp]: "Themal" => "Thermal"

File: Firmware/util.h [hunk 2]
```diff
@@ -106,8 +106,8 @@ typedef uint8_t CalibrationStatus;
 const CalibrationStatus CALIBRATION_STATUS_SELFTEST    = 0b00000001; // Selftest
 const CalibrationStatus CALIBRATION_STATUS_XYZ         = 0b00000010; // XYZ calibration
 const CalibrationStatus CALIBRATION_STATUS_Z           = 0b00000100; // Z calibration
-#ifdef TEMP_MODEL
-const CalibrationStatus CALIBRATION_STATUS_TEMP_MODEL  = 0b00001000; // Temperature model calibration
+#ifdef THERMAL_MODEL
+const CalibrationStatus CALIBRATION_STATUS_THERMAL_MODEL  = 0b00001000; // Thermal model calibration
```
Comment (wavexx) [inline Firmware/util.h]: In this case please re-align the equals

**Review by wavexx — CHANGES_REQUESTED** [review]
In addition to the previous comments, rhe PR needs another rebase to get the new REVO/REVO_HF_60W files and needs to rename them too from temp_mode->thermal_model.

The same is true for the REVO variants in Firmware/variants.

The file Firmware/variants/README.md also will need an update.

I repeated all the changes to the sources and I'm getting identical results :+1:

File: Firmware/ultralcd.h [hunk 3]
```diff
@@ -225,9 +225,9 @@ enum class WizState : uint8_t
     Selftest,       //!< self test
     Xyz,            //!< xyz calibration
     Z,              //!< z calibration
-#ifdef TEMP_MODEL
-    TempModel,      //!< Temp model calibration
-#endif //TEMP_MODEL
+#ifdef THERMAL_MODEL
+    ThermalModel,      //!< Thermal Model calibration
```
Comment (wavexx) [inline Firmware/ultralcd.h]: Alignment here

File: Firmware/menu.h [hunk 4]
```diff
@@ -35,8 +35,8 @@ extern uint8_t menu_depth;
 enum ESeriousErrors {
 	MENU_BLOCK_NONE                = 0,
 	MENU_BLOCK_THERMAL_ERROR       = 0x01,
-#ifdef TEMP_MODEL
-	MENU_BLOCK_TEMP_MODEL_AUTOTUNE = 0x02,
+#ifdef THERMAL_MODEL
+	MENU_BLOCK_THERMAL_MODEL_AUTOTUNE = 0x02,
```
Comment (wavexx) [inline Firmware/menu.h:39]: Also alignment here

File: Firmware/thermal_model.h [hunk 5]
```diff
@@ -6,27 +6,27 @@
...
+static const float THERMAL_MODEL_R_DEFAULT[THERMAL_MODEL_R_SIZE] PROGMEM = THERMAL_MODEL_DEF(Rv);
 
-namespace temp_model {
+namespace thermal_model {
 
 struct model_data
 {
     // temporary buffers
-    float dT_lag_buf[TEMP_MODEL_MAX_LAG_SIZE]; // transport delay buffer
+    float dT_lag_buf[THERMAL_MODEL_MAX_LAG_SIZE]; // transport delay buffer
```
Comment (wavexx) [inline Firmware/thermal_model.h:29]: Also alignment broken here and just below

**Comment by 3d-gussner** [comment]
@wavexx Can you please review again?
- rebased
- updated REVO variants and REVO tm
- updated alignment

File: Firmware/thermal_model.h [hunk 6]
```diff
@@ -39,7 +39,7 @@ struct model_data
     float C;                               // heatblock capacitance (J/K)
     float fS;                              // sim. 1st order IIR filter factor (f=100/27)
     uint16_t L;                            // sim. response lag (ms)
-    float R[TEMP_MODEL_R_SIZE];            // heatblock resistance for all fan levels (K/W)
+    float R[THERMAL_MODEL_R_SIZE];            // heatblock resistance for all fan levels (K/W)
```
Comment (wavexx) [inline Firmware/thermal_model.h]: This one is still missing

File: lang/po/Firmware_sv.po [hunk 7]
```diff
@@ -2626,7 +2626,7 @@ msgstr "Ett oväntat fel inträffade."
 #~ msgstr "TM autoinst fallerad"
 
 #~ msgid "Temp. model autotune"
-#~ msgstr "Temp modell autoinst"
+#~ msgstr "Thermal Modell autoinst"
```
Comment (wavexx) [inline lang/po/Firmware_sv.po]: "Temp. model" messages no longer exist, so this is kind of pointless. I would drop this, and in a new commit purge the obsolete translations.

Comment (3d-gussner) [reply lang/po/Firmware_sv.po]: Done

File: Firmware/thermal_model.h (hunk 6 above)
Comment (3d-gussner) [reply Firmware/thermal_model.h]: Done

**Review by wavexx — APPROVED** [review]
The last commit message seems like I'm your wife ;)

**Pull request merged**